lib/src/widget/nav_bar/widget/nav_bar_container.dart

```diff
@@ -0,0 +1,159 @@
+import 'package:flutter/material.dart';
+import 'package:xayn_design/src/widget/nav_bar/data/config_pair.dart';
+import 'package:xayn_design/src/widget/nav_bar/widget/nav_bar.dart';
+import 'package:xayn_design/xayn_design.dart';
+
+class NavBarContainer extends StatefulWidget {
+  final MaterialApp child;
+
+  const NavBarContainer({
+    Key? key,
+    required this.child,
+  }) : super(key: key);
+
+  @override
+  _NavBarContainerState createState() => _NavBarContainerState();
+
+  /// Call this, when you need to update current instance of the [NavBar]
+  /// f.e. [NavBarConfig] was changed for the screen
+  static void updateNavBar(BuildContext context) =>
+      context.dependOnInheritedWidgetOfExactType<_Updater>()!.data.updateBar();
+
+  /// This method used internally.
+  /// Will initiate walk through the children widgets,
+  /// find the latest instance of [NavBarConfig] and if it is not [null],
+  /// then will draw [NavBar] with it.
+  /// If not [NavBarConfig] found - [NavBar] will be not shown(removed)
+  ///
+  /// [goingBack] is a flag, that indicates that navigation flow going back.
+  /// when [true], then will ignore latest [NavBarConfig] (from screen that will be removed)
+  /// and will try to apply previous [NavBarConfig] (from screen that will be shown instead)
+  static void resetNavBar(
+    BuildContext context, {
+    required bool goingBack,
+  }) =>
+      context
+          .dependOnInheritedWidgetOfExactType<_Updater>()!
+          .data
+          .resetBar(goingBack);
+}
+
+class _NavBarContainerState extends State<NavBarContainer> {
+  ConfigPair? _configPair;
+
+  Linden get linden => UnterDenLinden.getLinden(context);
+
+  @override
+  Widget build(BuildContext context) {
+    final children = <Widget>[
+      Positioned.fill(child: widget.child),
+    ];
+    final navBarConfig = _configPair?.config;
+    if (navBarConfig != null) {
+      final bar = _buildBar(navBarConfig);
+      children.add(bar);
+    }
+
+    final stack = Stack(children: children);
+
+    final materialApp = MaterialApp(
+      home: stack,
+      theme: widget.child.theme,
+    );
+    return _Updater(
+      data: this,
+      child: materialApp,
+    );
+  }
```

Comment:
❓  Why is this building the MaterialApp ... IMHO this should not be the responsibility of the conatainer, it should only have another child (like a stack or whatever) and then do its magic by finding the Config and the NavBar

Reply:
Also I have some issues to understand the `_Updater` role, maybe we should have a call so I can understand this better

Reply:
We need this instance of `Material` to provide proper styling and themes for the widgets, that are living in `NavBarContainer`

Reply:
regarding `InheritedWidget` - we need to wrap our code into `InheritedWidget`.

And yes, feel free to drop me a msg regarding the call 😉